pkg/services/authn/clients/session.go

```diff
@@ -18,20 +20,25 @@ import (
 var _ authn.HookClient = new(Session)
 var _ authn.ContextAwareClient = new(Session)
 
-func ProvideSession(cfg *setting.Cfg, sessionService auth.UserTokenService, features *featuremgmt.FeatureManager) *Session {
+func ProvideSession(cfg *setting.Cfg, sessionService auth.UserTokenService,
+	features *featuremgmt.FeatureManager, anonDeviceService anonymous.Service) *Session {
 	return &Session{
-		cfg:            cfg,
-		features:       features,
-		sessionService: sessionService,
-		log:            log.New(authn.ClientSession),
+		cfg:               cfg,
+		features:          features,
+		sessionService:    sessionService,
+		log:               log.New(authn.ClientSession),
+		anonDeviceService: anonDeviceService,
+		tagDevices:        cfg.TagAuthedDevices,
```

Comment:
nit: no need to pass the specific config option here if we're passing the config itself.

Reply:
pre-optimization on the fact that clients should be 'stateless'

Reply:
`pre-optimization` 😈

Reply:
yup, can't deny it.